src/main/java/com/salesforce/dynamodbv2/mt/mappers/sharedtable/impl/MtAmazonDynamoDbBySharedTable.java

```diff
@@ -551,6 +591,121 @@ public UpdateItemResult updateItem(UpdateItemRequest updateItemRequest) {
+                + "managed on an mt-dynamo table builder, see <insert link to backup guide>");
+        }
+    }
+
+    @Override
+    public RestoreTableFromBackupResult restoreTableFromBackup(
+        RestoreTableFromBackupRequest restoreTableFromBackupRequest) {
+        if (backupManager.isPresent()) {
+
+            TenantTableBackupMetadata backupMetadata = ((MtSharedTableBackupManager) backupManager.get())
+                    .getTenantTableBackupFromArn(restoreTableFromBackupRequest.getBackupArn());
+            RestoreMtBackupRequest mtRestoreRequest = new RestoreMtBackupRequest(backupMetadata.getBackupName(),
+                new TenantTable(backupMetadata.getVirtualTableName(), backupMetadata.getTenantId()),
+                new TenantTable(restoreTableFromBackupRequest.getTargetTableName(), getMtContext().getContext()));
+            TenantRestoreMetadata restoreMetadata = backupManager.get().restoreTenantTableBackup(
+                mtRestoreRequest,
+                getMtContext());
+
+            return new RestoreTableFromBackupResult().withTableDescription(
+                mtTableDescriptionRepo.getTableDescription(restoreMetadata.getVirtualTableName()));
+        } else {
+            throw new ContinuousBackupsUnavailableException("Backups can only be created by configuring a backup "
+                + "managed on an mt-dynamo table builder, see <insert link to backup guide>");
+        }
+    }
+
+    @Override
+    public DescribeBackupResult describeBackup(DescribeBackupRequest describeBackupRequest) {
+        if (backupManager.isPresent()) {
+            if (getMtContext().getContextOpt().isPresent()) {
+                throw new UnsupportedOperationException("TODO: Implement tenant table backup describe");
+            }
+            MtBackupMetadata backupMetadata = getBackupManager().getBackup(describeBackupRequest.getBackupArn());
+            if (backupMetadata != null) {
+                return MtBackupAwsAdaptorKt.getBackupAdaptorSingleton().getDescribeBackupResult(backupMetadata);
+            } else {
+                throw new BackupNotFoundException("No backup with arn "
+                    + describeBackupRequest.getBackupArn() + "found");
+            }
+        } else {
+            throw new ContinuousBackupsUnavailableException("Backups can only be created by configuring a backup "
+                + "managed on an mt-dynamo table builder, see <insert link to backup guide>");
+        }
+    }
+
+    @Override
+    public CreateBackupResult createBackup(CreateBackupRequest createBackupRequest) {
+        if (backupManager.isPresent()) {
+            Preconditions.checkNotNull(createBackupRequest.getBackupName(), "Must pass backup name.");
+            Preconditions.checkArgument(createBackupRequest.getTableName() == null,
+                "Multitenant backups cannot backup individual tables, table name arguments are disallowed.");
```

Comment:
It seems like this should be a separate method that does not implement the `AmazonDynamoDB` interface, since it's creating a backup for all physical tables managed by this shared table instance. Don't feel super strongly about it though, since we can change that later (since ZOS will likely be the only consumer for a while).

Reply:
right, I remember that conversation. Sure, I can move the backup hook to a public method of `MtAmazonDynamoDbBySharedTable` that ZOS can call directly, instead of abusing this API.